images/cluster/build.sh

```diff
@@ -14,26 +14,167 @@
-
-KUBE_VERSION="v1.24.2"
-while [[ $# -gt 0 ]]; do
-  arg="$1"
-  case "${arg}" in
-  --kube-version | --kube-version=*)
-    [[ "${key#*=}" != "${key}" ]] && KUBE_VERSION="${key#*=}" || { KUBE_VERSION="${2}" && shift; }
-    shift
-    ;;
-  *)
-    args+=("${arg}")
-    ;;
-  esac
-  shift
-done
-
-docker build \
-  "${args[@]}" \
-  --build-arg kube_version="${KUBE_VERSION}" \
-  -f "${DIR}/Dockerfile" \
-  "${DIR}/../.."
+DRY_RUN=false
+PUSH=false
+IMAGES=()
+EXTRA_TAGS=()
+PLATFORMS=()
+VERSION=""
+KUBE_VERSIONS=()
+
+function usage() {
+  echo "Usage: ${0} [--help] [--version <version>] [--kube-version <kube-version> ...] [--image <image> ...] [--extra-tag <extra-tag> ...] [--platform <platform> ...] [--push] [--dry-run]"
+  echo "  --version <version> is kwok version, is required"
+  echo "  --kube-version <kube-version> is kubernetes version, is required"
+  echo "  --image <image> is image, is required"
+  echo "  --extra-tag <extra-tag> is extra tag"
+  echo "  --platform <platform> is multi-platform capable for image"
+  echo "  --push will push image to registry"
+  echo "  --dry-run just show what would be done"
+}
+
+function args() {
+  local arg
+  while [[ $# -gt 0 ]]; do
+    arg="$1"
+    case "${arg}" in
+    --kube-version | --kube-version=*)
+      [[ "${arg#*=}" != "${arg}" ]] && KUBE_VERSIONS+=("${arg#*=}") || { KUBE_VERSIONS+=("${2}") && shift; }
+      shift
+      ;;
+    --version | --version=*)
+      [[ "${arg#*=}" != "${arg}" ]] && VERSION="${arg#*=}" || { VERSION="${2}" && shift; }
+      shift
+      ;;
+    --image | --image=*)
+      [[ "${arg#*=}" != "${arg}" ]] && IMAGES+=("${arg#*=}") || { IMAGES+=("${2}") && shift; }
+      shift
+      ;;
+    --extra-tag | --extra-tag=*)
+      [[ "${arg#*=}" != "${arg}" ]] && EXTRA_TAGS+=("${arg#*=}") || { EXTRA_TAGS+=("${2}") && shift; }
+      shift
+      ;;
+    --platform | --platform=*)
+      [[ "${arg#*=}" != "${arg}" ]] && PLATFORMS+=("${arg#*=}") || { PLATFORMS+=("${2}") && shift; }
+      shift
+      ;;
+    --push | --push=*)
+      [[ "${arg#*=}" != "${arg}" ]] && PUSH="${arg#*=}" || PUSH="true"
+      shift
+      ;;
+    --dry-run | --dry-run=*)
+      [[ "${arg#*=}" != "${arg}" ]] && DRY_RUN="${arg#*=}" || DRY_RUN="true"
+      shift
+      ;;
+    --help)
+      usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown argument: ${arg}"
+      usage
+      exit 1
+      ;;
+    esac
+  done
+
+  if [[ "${VERSION}" == "" ]]; then
+    echo "--version is required"
+    usage
+    exit 1
+  fi
+
+  if [[ "${#KUBE_VERSIONS}" -eq 0 ]]; then
+    echo "--kube-version is required"
+    usage
+    exit 1
+  fi
+
+  if [[ "${#IMAGES}" -eq 0 ]]; then
+    echo "--image is required"
+    usage
+    exit 1
+  fi
+}
+
+function dry_run() {
+  echo "${@}"
+  if [[ "${DRY_RUN}" != "true" ]]; then
+    eval "${@}"
+  fi
+}
+
+function main() {
+  local extra_args
+  local suffix
+  if [[ "${#PLATFORMS}" -ne 0 ]]; then
+    export DOCKER_CLI_EXPERIMENTAL=enabled
```

Comment:
do we actually need this?
I thought we should be able to use buildkit without needing to set `DOCKER_CLI_EXPERIMENTAL`

Reply:
`docker buildx` requires this to work.